libdnf5/module/module_sack.cpp

```diff
@@ -459,12 +463,12 @@ void ModuleSack::Impl::enable_dependent_modules() {
 }
 
 
-std::pair<std::vector<std::vector<std::string>>, ModuleSack::ModuleErrorType> ModuleSack::Impl::module_solve(
-    std::vector<ModuleItem *> module_items) {
-    std::vector<std::vector<std::string>> problems;
+std::pair<std::vector<std::vector<std::tuple<ProblemRules, Id, Id, Id, std::string>>>, GoalProblem>
+ModuleSack::Impl::module_solve(std::vector<ModuleItem *> module_items) {
```

Comment:
I believe that using a reference would be better.

`ModuleSack::Impl::module_solve(std::vector<ModuleItem *> & module_items)`

Reply:
Agreed, fixed.